### ISSUE TITLE
Support for Mac OS X 10.6.8

### ISSUE THREAD

**Issue by porg** [Syncplay/syncplay#173]
I'd like to run SyncPlay on an old MacBook Pro running Mac OS X 10.6.8

http://syncplay.pl/guide/install/ does not explicitly state any system requirements, so I just tried running the newest stable version 1.5.1 and 1.5.2 beta 3. I have had updated VLC to the newest available version 2.2.8.

The SyncPlay GUI app launches, the OS does not give any "not supported" feedback. But the app seems to fail silently.

**When looking into Console.app, I see this output from SyncPlay:**
```
Could not import GUI libraries. If you do not have PySide installed then you will need to install it  for the GUI to work.
Hostname can't be empty!
Some necessary arguments are missing, refer to --help
```
**My questions:**
1. What is the highest SyncPlay version supported for Mac OS X 10.6.8?
2. Where can I download legacy versions?
3. What could I attempt to get the newest SyncPlay running on Mac OX X 10.6.8?

**Comment by porg** [comment]
I just read through the release history, and found that only the very recent versions 1.5.x added proper Mac support. With Mac support being very young for this software, I hope that my feedback for the legacy Mac OS X 10.6, has good chances to be seen and maybe fixed.

I am optimistic. Would love to use my old hardware for that purpose.

**Comment by alby128** [comment]
So, let me wrap this for you. Firstly, as far as I know, there are no technical incompatibilities that prevent running the latest version of Syncplay on 10.6.8. Personally, I have tested it down to 10.7.x and it works, so I am not 100% sure, but I am confident enough that it should work. This means that all the changes that we did to add Mac "proper" support should work also on 10.6.8.

Secondly, the reason why we are releasing packed .dmg only compatible with 10.10+ is entirely due to how we automatically create those. Not many Syncplay's active developers have a Mac, so we employ an automatic platform (Travis CI) to generate these .dmg that are updated to the latest commit and can be directly distributed to the users. Unfortunately, Travis CI does not support systems older than Yosemite.

Thirdly, before introducing Travis CI support I used to manually freeze and pack the code on my 10.7 machine, so packed .dmg can in principle support older operative systems provided that someone generates them manually. At this moment, unfortunately, I cannot prepare a version compatible with Snow Leopard for you simply because I do not have a working copy of 10.6 installed.

In summary, there is no need for you to download a legacy version, you just need to set up your system right for executing the current codebase from sources. To help you with that, I advice you to read my blog post with detailed manual installation instructions [here](http://www.albertosottile.it/syncplay/2017/09/08/install-syncplay-on-macos-10-8/) (forget about the title). I suggest you avoid miniconda and use homebrew, it should be able to provide older versions of the dependencies. Let me know about your progress and feel free to comment here if you encounter any error.

**Comment by alby128** [comment]
Side note: support for legacy Macs will be broken when we migrate to Qt 5, which, as far as I know, are not supported by these systems. Also, for @Et0h : if you think this is worthwhile, I could prepare a frozen .dmg of 1.5.1 (for example) that runs on macOS 10.7+ and we could put that on the website as a sort of "legacy" Mac release that stays there and is not updated at all. The reason why I did not propose this before is that I thought there was no interest in such old versions of macOS, but it appears I might have been wrong.

**Comment by Et0h** [comment]
@alby128 A build for macOS 10.7 sounds good, but as 1.5.2 fixes a bug with file duration for mpv it's probably best waiting for that to be released (should be soon) and use that as the base.

@porg It'd be helpful to know if alby128's instructions work exactly as stated for macOS 10.6, because if that is the case we can link to it from our websute for 10.6 users.

**Comment by porg** [comment]
@alby128 Mille grazie for getting involved!

@Et0h and @alby128: I looked at the instructions of the link provided by @alby128.

Installing homebrew on that old system was not working, due to curl and git being to old, thus the fetch/update routines fail. I gave up after many hours of trying workarounds. There sure is a possibility to get brew running on 10.6.8, but before investing into further attempts, I'd like to ask for another approach:

I still have a fink on that machine. What steps do I need to take to run it with fink?

And last but not least: I'm not the most tech-savvy user (anymore), should I fail with the package managers and runnung from tarball source code, I'd appreciate a .DMG of a frozen version, 1.5.2, and would be more than glad, should that give me another few years using the software as-is.

**Comment by alby128** [comment]
Nevermind, I installed a VM with 10.6.8 Server and generated the dmg myself. @porg you can download it from [here](https://bintray.com/alby128/Syncplay/download_file?file_path=Syncplay_1.5.1_macOS_10.6.dmg) and install it as described in the included README file. It should be compatible with all the macOS releases from 10.6.8 to 10.12.6 (I did not test Syncplay on High Sierra yet). Please, let me know if you find any issues with this build. 

@Et0h once these tests are finished and 1.5.2 is finalized, I will generate a "legacy" dmg using this VM, so that we will have a static version compatible with older Macs on the website.

**Comment by alby128** [comment]
@porg Did you test the package that I linked in the previous comment? It should be compatible with 10.6.8. I am waiting for your tests to proceed with this issue. Thanks.

**Comment by porg** [comment]
Yes, the DMG with Syncplay 1.5.1 ran without any issues on macOS 10.6.8.

P.S.: I had prepared my response already on the next day from the machine where I tested it but seem to have forgotten to submit it. Sorry that you needed to run after me. Proactive engagement is the least I could do for receiving such a cool software for my old machine! I really merely forgot to submit. Sorry! Please also ping me when the "feature final version 1.5.2" is ready for macOS 10.6.8. I'd be glad to test it for you.

**Comment by Et0h** [comment]
@albertosottile 1.5.2 is now "feature final" so ready for a legacy build.

**Comment by albertosottile** [comment]
A legacy dmg updated to the latest commit (so 1.5.2 final) is now available on the Syncplay's Bintray page ([direct link](https://bintray.com/syncplay/Syncplay/download_file?file_path=Syncplay_1.5.2_macOS_legacy.dmg)). 

It should work on macOS 10.6.8+ and I tested it on my 10.6.8 VM and my 10.12.6 systems. Maybe @porg could test it again but, as far as I know, it is ready for release. 

@Et0h I set milestone as 'macOS_legacy' as we agreed. Once the legacy .dmg is put on the website, this issue can be closed.

**Comment by Et0h** [comment]
albertosottile's legacy version Syncplay designed for macOS 10.6.8+ is now listed at http://syncplay.pl/syncplay-1-5-2/ - if anyone has a problem with it then please let us know.
